File: src/main/java/com/wisepc/wisepc/configurator/http/ConfiguratorController.java
package com.wisepc.wisepc.configurator.http;

import com.wisepc.wisepc.configurator.ConfiguratorService;
import com.wisepc.wisepc.configurator.model.ConfiguratorUpdateRequest;
import com.wisepc.wisepc.session.SessionService;

import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import lombok.RequiredArgsConstructor;

@RestController
@RequestMapping(value = "/configurator")
@RequiredArgsConstructor
class ConfiguratorController {
    private final SessionService sessionService;
    private final ConfiguratorService configuratorService;

    @PostMapping(value = "/update-configuration")
    public void updateConfiguration(@RequestHeader(value = "sessionToken", required = true) String sessionToken,
                                    @RequestBody ConfiguratorUpdateRequest configuratorUpdateRequest) {

        configuratorService.updateOrCreateSession(sessionToken, configuratorUpdateRequest);
    }
}

File: src/main/java/com/wisepc/wisepc/configurator/http/ConfiguratorController.kt
package com.wisepc.wisepc.configurator.http

import com.wisepc.wisepc.configurator.ConfiguratorService
import com.wisepc.wisepc.configurator.dto.ConfiguratorUpdateRequest
import com.wisepc.wisepc.session.SessionService
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(value = ["/configurator"])
class ConfiguratorController(
        private val sessionService: SessionService,
        private val configuratorService: ConfiguratorService) {

    @PostMapping(value = ["/update-configuration"], consumes = [MediaType.APPLICATION_JSON_VALUE])
    fun updateConfiguration(@RequestHeader(value = "sessionToken", required = true) sessionToken: String?,
                            @RequestBody configuratorUpdateRequest: ConfiguratorUpdateRequest?) {
        configuratorService.updateOrCreateSession(sessionToken, configuratorUpdateRequest)
    }
}